src/main/java/co/inventorsoft/mailsecurity/repositories/EmailRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,127 @@
+package co.inventorsoft.mailsecurity.repositories;
+
+import co.inventorsoft.mailsecurity.models.Email;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class EmailRepositoryJdbcImpl implements EmailRepository {
+
+    private Connection connection;
+
+    public EmailRepositoryJdbcImpl(DataSource dataSource) {
```

Comment:
It is good that you know how to work with DataSource, but idea of task was to do everything manually, so you could understand how it works under the hood.